File: app/src/main/java/kandroid/data/MapInfoData.java
package kandroid.data;

import kandroid.observer.POJO;
import kandroid.utils.Identifiable;

public class MapInfoData extends Data<MapInfoData.ApiMapInfo> implements Identifiable {

    public int getMapID() {
        return data.api_id;
    }

    public Start2Data.MasterMapInfoData getMasterMapInfo() {
        return KCDatabase.getInstance().master.masterMapInfoData.get(getMapID());
    }

    public int getMapAreaID() {
        return getMasterMapInfo().getMapAreaID();
    }

    public int getMapInfoID() {
        return getMasterMapInfo().getMapInfoID();
    }

    public String getName() {
        return getMasterMapInfo().getName();
    }

    public int getDifficulty() {
        return getMasterMapInfo().getDifficulty();
    }

    public String getOperationName() {
        return getMasterMapInfo().getOperationName();
    }

    public String getInformation() {
        return getMasterMapInfo().getInformation();
    }

    public int getRequiredDefeatedCount() {
        return getMasterMapInfo().getRequiredDefeatedCount();
    }

    public boolean isCleared() {
        return data.api_cleared != 0;
    }

    public int getCurrentDefeatedCount() {
        return data.api_defeat_count;
    }

    public int getMapHPCurrent() {
        return data.api_eventmap.api_now_maphp;
    }

    public int getMapHPMax() {
        return data.api_eventmap.api_max_maphp;
    }

    public int getEventDifficulty() {
        return data.api_eventmap.api_selected_rank;
    }

    public int getGaugeType() {
        return data.api_eventmap.api_gauge_type;
    }

    @Override
    public int getID() {
        return data.api_id;
    }

    public static class ApiMapInfo extends POJO {
        public int api_id;
        public int api_cleared;
        public int api_defeat_count;
        public int api_exboss_flag;
        public ApiEventmap api_eventmap = new ApiEventmap();

        private class ApiEventmap {
            public int api_now_maphp;
            public int api_max_maphp;
            public int api_state;
            public int api_selected_rank;
            public int api_gauge_type;
            public int api_airbase_enabled;
        }
    }
}

File: app/src/main/java/kandroid/data/MapInfoData.kt
package kandroid.data

import kandroid.data.Start2Data.MasterMapInfoData
import kandroid.data.JsonWrapper
import kandroid.utils.Identifiable
import kandroid.utils.json.get
import kandroid.utils.json.int

class MapInfoData : ApiWrapper(), Identifiable {

    val mapID: Int get() = data["api_id"].int()

    val masterMapInfo: MasterMapInfoData get() = KCDatabase.master.masterMapInfoData[mapID]!!

    val mapAreaID: Int get() = masterMapInfo.mapAreaId

    val mapInfoID: Int get() = masterMapInfo.mapInfoId

    val name: String get() = masterMapInfo.name

    val difficulty: Int get() = masterMapInfo.difficulty

    val operationName: String get() = masterMapInfo.operationName

    val information: String get() = masterMapInfo.information

    val requiredDefeatedCount: Int get() = masterMapInfo.requiredDefeatedCount

    val isCleared: Boolean get() = data["api_cleared"].int() != 0

    val currentDefeatedCount: Int get() = data["api_defeat_count"].int()

    val mapHPCurrent: Int get() = data["api_eventmap"]["api_now_maphp"].int()

    val mapHPMax: Int get() = data["api_eventmap"]["api_max_maphp"].int()

    val eventDifficulty: Int get() = data["api_eventmap"]["api_selected_rank"].int()

    val gaugeType: Int get() = data["api_eventmap"]["api_gauge_type"].int()

    override val id: Int get() = mapID
}